docs/testing.md

```diff
@@ -0,0 +1,17 @@
+Testing
+=======
+
+Running `make test` runs most of the integrated tests. You should have
+`valgrind` installed.
+
+Running `make pre-push` runs the above builds and tests in different configurations: GCC,
+clang, and with ASAN enabled.
```

Comment:
Any special reason to move this to next line?

Reply:
Do you mean, why a new paragraph? Only so it's clearly separated from "make test" description.

Reply:
Ohh sorry I misunderstood it. Ignore me.